styleguide/screens/Patterns/Calendar/Calendar.js

```diff
@@ -0,0 +1,321 @@
+          <T elm="p" className={m.mtr}>
+            The <C>CalendarMonth</C> column heading format can be controlled using the
+            <C>columnHeadingProps</C> and <C>format</C> props.
+          </T>
+          <Specimen
+            classNames={{
+              root: m.mtr,
+              specimenContainer: m.par,
+            }}
+            code={dedent`
+              <CalendarMonth columnHeadingProps={{ format: 'dddd' }} />
+            `}
+          >
+            <CalendarMonth
+              columnHeadingProps={{
+                format: 'dddd',
+              }}
+            />
+          </Specimen>
+
+          <H level={3} className={m.mtLgIi}>Day Picker</H>
+          <T elm="p" className={m.mtr}>
+            Like the calendar month, but allows a day to be selected when clicked or touched this is
+            indicated by a filled square. Using the left and right chevrons you can navigate
+            through the months.
+          </T>
+          <Specimen
+            classNames={{
+              root: m.mtr,
+              specimenContainer: m.par,
+            }}
+            code={dedent`
+                <DayPicker
+                  month={month}
+                  onInteraction={this.handleInteraction}
+                  onMonthChange={this.handleMonthChange}
+                  dayProps={{ getDayState: this.getDayState }}
+                />
+            `}
+          >
+            <DayPicker
+              name="dayPicker"
+              month={dayPicker.month}
+              onInteraction={this.handleDayPickerInteraction}
+              onMonthChange={this.handleDayPickerMonthChange}
+              dayProps={{ getDayState: this.getDayState }}
+            />
+          </Specimen>
+
+          <H level={3} className={m.mtLgIi}>Day Range Picker</H>
+          <T elm="p" className={m.mtr}>
+            Clicking a date selects a start date moving the mouse into the future indicates the
+            range using a row or block of filled squares. Clicking again selects the end date.
+          </T>
+          <Specimen
+            classNames={{
+              root: m.mtr,
+              specimenContainer: m.par,
+            }}
+            code={dedent`
+              <DayRangePicker
+                startDate={dayRangePicker.startDate}
+                endDate={dayRangePicker.endDate}
+                selectDate={dayRangePicker.selectDate}
+                month={dayRangePicker.month}
+                onInteraction={this.handleDayRangeInteraction}
+                onMonthChange={this.handleDayRangeMonthChange}
+              />
+            `}
+          >
+            <DayRangePicker
+              startDate={dayRangePicker.startDate}
+              endDate={dayRangePicker.endDate}
+              selectDate={dayRangePicker.selectDate}
+              month={dayRangePicker.month}
+              onInteraction={this.handleDayRangeInteraction}
+              onMonthChange={this.handleDayRangeMonthChange}
```

Comment:
I tried this one out but it's not working as expected. Changing month and then clicking on a day of that month brings me back to the selection I already made in the previous month.

Reply:
I agree this behaviour could be improved, let's address this separately.